File: LLM-Translations/GPT-4/avatar/Java/C/atcoder_ABC127_F.c
#include <stdio.h>
#include <stdlib.h>

int comp(const void * a, const void * b) {
   return (*(int*)b - *(int*)a);
}

int main() {
    int t, in, i;
    long sumMin = 0, sumMax = 0, sumb = 0;
    int min = 0;
    int pqmin[100000], pqmax[100000];
    int sizeMin = 0, sizeMax = 0;
    
    scanf("%d", &t);
    while (t--) {
        int type;
        scanf("%d", &type);
        if(type == 2) {
            long ans = min;
            ans *= sizeMin;
            ans -= sumMin;
            long ans1 = min;
            ans1 *= sizeMax;
            ans1 = sumMax - ans1;
            printf("%d %ld\n", min, ans + ans1 + sumb);
        } else {
            int b;
            scanf("%d %d", &in, &b);
            sumb += b;
            if (in > min) {
                pqmax[sizeMax++] = in;
                sumMax += in;
            } else {
                pqmin[sizeMin++] = in;
                sumMin += in;
            }
            qsort(pqmin, sizeMin, sizeof(int), comp);
            qsort(pqmax, sizeMax, sizeof(int), comp);
            if (sizeMin > sizeMax) {
                sumMax += pqmin[sizeMin-1];
                sumMin -= pqmin[sizeMin-1];
                pqmax[sizeMax++] = pqmin[--sizeMin];
            }
            if (sizeMin < sizeMax) {
                sumMax -= pqmax[sizeMax-1];
                sumMin += pqmax[sizeMax-1];
                pqmin[sizeMin++] = pqmax[--sizeMax];
            }
            min = pqmin[sizeMin-1];
        }
    }
    return 0;
} // //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_ABC127_F.go
Go
package main

import (
	"bufio"
	"container/heap"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type MaxHeap []int
type MinHeap []int

func (h MaxHeap) Len() int           { return len(h) }
func (h MaxHeap) Less(i, j int) bool { return h[i] > h[j] }
func (h MaxHeap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
func (h *MaxHeap) Push(x interface{}) {
	*h = append(*h, x.(int))
}
func (h *MaxHeap) Pop() interface{} {
	old := *h
	n := len(old)
	x := old[n-1]
	*h = old[0 : n-1]
	return x
}

func (h MinHeap) Len() int           { return len(h) }
func (h MinHeap) Less(i, j int) bool { return h[i] < h[j] }
func (h MinHeap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
func (h *MinHeap) Push(x interface{}) {
	*h = append(*h, x.(int))
}
func (h *MinHeap) Pop() interface{} {
	old := *h
	n := len(old)
	x := old[n-1]
	*h = old[0 : n-1]
	return x
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	t, _ := strconv.Atoi(scanner.Text())

	pqmax := &MaxHeap{}
	heap.Init(pqmax)
	pqmin := &MinHeap{}
	heap.Init(pqmin)

	sumMin, sumMax, sumb := 0, 0, 0
	min := 0
	for i := 0; i < t; i++ {
		scanner.Scan()
		s := strings.Split(scanner.Text(), " ")
		if s[0] == "2" {
			ans := min * pqmin.Len()
			ans -= sumMin
			ans1 := min * pqmax.Len()
			ans1 = sumMax - ans1
			fmt.Println(min, ans+ans1+sumb)
		} else {
			in, _ := strconv.Atoi(s[1])
			sumb += int(s[2])
			if in > min {
				heap.Push(pqmax, in)
				sumMax += in
			} else {
				heap.Push(pqmin, in)
				sumMin += in
			}
			if pqmin.Len() > pqmax.Len() {
				sumMax += (*pqmin)[0]
				sumMin -= (*pqmin)[0]
				heap.Push(pqmax, heap.Pop(pqmin))
			}
			if pqmin.Len() < pqmax.Len() {
				sumMax -= (*pqmax)[0]
				sumMin += (*pqmax)[0]
				heap.Push(pqmin, heap.Pop(pqmax))
			}
			min = (*pqmin)[0]
		}
	}
}
// //End of Code


